client/ayon_core/pipeline/publish/lib.py

```diff
@@ -945,3 +945,19 @@ def get_publish_instance_families(instance):
         families.discard(family)
     output.extend(families)
     return output
+
+
+def register_representation(instance, traits_data):
+    """Register representation on instance.
+
+    Args:
+        instance (pyblish.api.Instance): Instance on which should be
+            representation registered.
+        traits_data (dict): Representation traits data.
```

Comment:
Can we provide better type hints than `dict` for this. First off, why can't we just have an actual `Traits` instance to pass in so we're correctly typing stuff. Or worst case scenario if there's so much need to be working with `dict` can we use `TypedDict` type hints to define what keys should exist, and what types those values are for those keys, etc.?

Reply:
`traits_data` is basically dict, but in OpenAssetIO world it is actually `TraitsData` type - here,  I am trying to avoid pulling in OpenAssetIO dependency into common lib file. How this work:

```python
# pseudo code
from openassetio.trait import TraitsData
from ayon_core.pipeline.traits import SomeTrait, OtherTrait


traits_data = TraitsData()

# now you can set traits data like so
SomeTrait(traits_data).setSomeData("foo")
OtherTrait(traits_data).setOther("bar")

# to check if trait is set on the data
if SomeTrait.isImbuedTo(traits_data):
    print("there is SomeTrait in traits_data")

# or it can be done like
some_trait = SomeTrait(traits_data)
if some_trait.isImbued():
    print("there is SomeTrait in traits_data")

# you can also check if the traits is in traits_data like
if traits_data.hasTrait(SomeTrait.kId):
	print("there is SomeTrait in traits_data")
```
Now, traits_data is something that will be serialized to json and stored in db (replacing representations in the future)

Reply:
Also I am thinking if it is good idea to reuse the `representations` key in an instance as I am doing here. Maybe in that case typing it explicitly to `TraitsData` would be better.